lib/scope.js

```diff
@@ -450,13 +440,25 @@ AST_Toplevel.DEFMETHOD("mangle_names", function(options){
             to_mangle.push(node.definition());
             return;
         }
-    });
-    this.walk(tw);
+    }));
     to_mangle.forEach(function(def){ def.mangle(options) });
 
     if (options.cache) {
         options.cache.cname = this.cname;
     }
+
+    function push_symbols(dict) {
+        var a = [];
+        dict.each(function(symbol) {
+            if (options.reserved.indexOf(symbol.name) < 0) {
+                a.push(symbol);
+            }
+        });
+        a.sort(function(m, n) {
+            return n.references.length - m.references.length;
```

Comment:
```patch
+        a.sort(function(m, n) {
+            return n.references.length - m.references.length;
```

I vaguely recall seeing that mangle approach attempted a couple years back and abandoned due to gzip results. I could be mistaken.